parcels/field.py

```diff
@@ -55,7 +55,8 @@ class Field(object):
     def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=None, mesh='flat',
                  fieldtype=None, transpose=False, vmin=None, vmax=None, time_origin=None,
                  interp_method='linear', allow_time_extrapolation=None, time_periodic=False, **kwargs):
-        self.name = name
+        self.name = list(name)[0] if isinstance(name, dict) else name
+        self.filebuffername = name[list(name)[0]] if isinstance(name, dict) else name
```

Comment:
It's quite hard to follow. Why is `name` a dict?
If you make `name` a tuple,  you can do
```
if not isinstance(name, tuple):
    self.name = name
    self.filebuffername = name
else:
   self.name, self.filebuffername = name
```
or as you did
```
self.name = name[0] if isinstance(name, tuple) else name
self.filebuffername = name[1] if isinstance(name, tuple) else name
```